src/connect.ts

```diff
@@ -72,33 +127,57 @@ export class ConnectQOS {
   }
 
   shouldThrottleRequest(req: IncomingMessage|Http2ServerRequest): BadActorType|boolean {
+    // do not track much less block local addresses
+    if (this.#exemptLocalAddress && isLocalAddress(req?.socket?.remoteAddress || '')) return false;
+
     this.#metrics.trackRequest(req);
 
-    if (this.tooBusy === true) {
-      const badActor = this.#metrics.isBadActor(req);
-      if (badActor) return badActor; // contains reason for bad actor
-      if (this.#userLag && this.lag >= this.#userLag) {
-        return BadActorType.userLag; // yes, throttle possibly innocent user
-      }
-    }
-  
+    // user lag takes priority even if no history is established to permit healthy recovery
+    if (this.lag >= this.#userLag) return BadActorType.userLag;
```

Comment:
I don't understand why userLag is checked first here - wouldn't that result in undercounting badHost/badIP traffic during attacks?

Reply:
User lag has always taken priority, but for sake of reporting statistics I can see why this change might be undesirable. I'll revisit.